jscomp/bsb/bsb_build_util.ml

```diff
@@ -229,5 +232,31 @@ let rec walk_all_deps_aux
     
 
 let walk_all_deps dir cb = 
-  let visited = String_hashtbl.create 0 in 
+  let visited = String_hashtbl.create 0 in
+  Bsb_log.info "@{<info>walk_all_deps dir:@} %s @." dir;
   walk_all_deps_aux visited [] true dir cb 
+
+let build_artifacts_dir = ref None
+
+let get_build_artifacts_location cwd =
+  let () = Bsb_log.info "@{<info>get_build_artifacts_location cwd:@} %s @." cwd in
+  let () = Bsb_log.info "@{<info>get_build_artifacts_location cur__root:@} %s @." (Sys.getenv "cur__root") in
+  let bad = match !build_artifacts_dir with
+  | Some(x) -> x
+  | None -> "not_set" in
+  let () = Bsb_log.info "@{<info>build_artifacts_dir:@} %s @." bad in
+  (* If the project's parent folder is not node_modules, we know it's the top level one. *)
+  if Sys.getenv "cur__root" = cwd then 
```

Comment:
I think getenv might throw an exception

Reply:
Yes, this is currently "safe" because we are probable in the `esy` environment. But this is mostly a placeholder for something like a flag.